### DISCUSSION TITLE
option to refetch only first group of useInfinteQuery results

### DISCUSSION THREAD

**Discussion by skulegirl** [TanStack/query#875] [General]
With useInifiniteQuery, currently if I have already loaded N pages, then calling refetch will result in N API calls to refetch all the data. I understand that this is by design.

For a react native app, I want a pull-from-top refresh to continue to display only the first page while it is refreshed, but remove all subsequent pages from the cache and not refetch them unless the fetchMore method is called again. Can we add a way to reset the number of loaded groups in the useInfiniteQuery cache data? I've tried using queryCache.setQueryData to delete all but the first group of data, but it's not sufficient as it still seems to know that more groups were previously fetched and then proceeds to refetch them.

**Comment by dpyzo0o** [comment]
Anyone has ideas?

**Comment by boschni** [comment]
This issue of remembering previous pages after a `setQueryData` would probably be fixed with https://github.com/tannerlinsley/react-query/pull/914 .

**Comment by vesparny** [comment]
@skulegirl did you find a solution by chance? I am struggling as well with the same issue.

**Comment by seanmiddaugh** [comment]
This solution worked well for me:

```
queryClient.setQueryData(['images', postId], data => ({ 
   pages: data.pages.slice(0, 1), 
   pageParams: data.pageParams.slice(0, 1)
}));
```

When I pull to refresh my RN FlatList I only wanted the first page to be refreshed/returned. I call this before I call `refetch`. Took me awhile to find this, hopefully it helps someone out.
